thread.c

```diff
@@ -564,11 +680,9 @@ void dispatch_conn_new(int sfd, enum conn_states init_state, int event_flags,
         return;
     }
 
-    int tid = (last_thread + 1) % settings.num_threads;
-
-    LIBEVENT_THREAD *thread = threads + tid;
+    LIBEVENT_THREAD *thread = NULL;
 
-    last_thread = tid;
+    select_thread(sfd, &thread);
```

Comment:
This is a little nitpicky but this routine makes the common case convoluted and with too much overhead:
- function call, which then checks for a flag, which then makes a function call (common case! most users will not be using this feature).
- select_thread() has an out: label with a handler that checks and runs `select_next_thread`, but that's only used once? another time `select_next_thread` is called explicitly, then sent to out:. Should at least be consistent.

Reply:
Agree, Once I rewrite that function, hopefully this comment will get addressed.